### PR TITLE
Fixes an issue with FeatureSIMD=0 and ArmBase.Yield

### LINKED ISSUE #62117: Test failure: JIT/HardwareIntrinsics/Arm/ArmBase/Yield_r* on Arm64
Run: [runtime-coreclr jitstress-isas-arm 20211127.1](https://dev.azure.com/dnceng/public/_build/results?buildId=1487833&view=ms.vss-test-web.build-test-results-tab&runId=42539954&paneView=debug&resultId=100409)

Failed test:
```

CoreCLR OSX arm64 Checked jitstress_isas_nosimd @ OSX.1100.ARM64.Open

- JIT/HardwareIntrinsics/Arm/ArmBase/Yield_r/Yield_r.sh
- JIT/HardwareIntrinsics/Arm/ArmBase/Yield_ro/Yield_ro.sh

CoreCLR Linux arm64 Checked jitstress_isas_nosimd @ (Ubuntu.1804.Arm64.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8-20210531091519-97d8652

- JIT/HardwareIntrinsics/Arm/ArmBase/Yield_r/Yield_r.sh
- JIT/HardwareIntrinsics/Arm/ArmBase/Yield_ro/Yield_ro.sh

```
**Error message：**
```
Return code:      1
Raw output file:      /tmp/helix/working/BA120941/w/9C6008B1/uploads/Reports/JIT.HardwareIntrinsics/Arm/ArmBase/Yield_r/Yield_r.output.txt
Raw output:
BEGIN EXECUTION
/tmp/helix/working/BA120941/p/corerun -p System.Reflection.Metadata.MetadataUpdater.IsSupported=false Yield_r.dll ''
Expected: 100
Actual: 0
END EXECUTION - FAILED
Test Harness Exitcode is : 1
To run the test:

set CORE_ROOT=/tmp/helix/working/BA120941/p
/private/tmp/helix/working/BA120941/w/9C6008B1/e/JIT/HardwareIntrinsics/Arm/ArmBase/Yield_r/Yield_r.sh
Expected: True
Actual:   False


Stack trace
   at JIT_HardwareIntrinsics._Arm_ArmBase_Yield_r_Yield_r_._Arm_ArmBase_Yield_r_Yield_r_sh()
```

### PR DESCRIPTION
**Pull request by echesakov**
The issue is that ` ArmBase.Yield` is not expanding as intrinsic when `FeatureSIMD=0`.
This shouldn't be a case.

In addition, this marks the intrinsic with `HW_Flag_NoFloatingPointUsed`.

Fixes https://github.com/dotnet/runtime/issues/62117

### CODE REVIEW

**Comment by echesakovMSFT** [comment]
@dotnet/jit-contrib @tannergooding PTAL

**Review by BruceForstall — APPROVED** [review]

File: src/coreclr/jit/hwintrinsicarm64.cpp [hunk 1]
```diff
@@ -308,8 +308,19 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
...
+    // The vast majority of "special" intrinsics are Vector64/Vector128 methods.
+    // The only exception is ArmBase.Yield which should be treated differently.
+    if (intrinsic == NI_ArmBase_Yield)
+    {
+        assert(sig->numArgs == 0);
+        assert(JITtype2varType(sig->retType) == TYP_VOID);
+        assert(simdSize == 0);
+
+        return gtNewScalarHWIntrinsicNode(TYP_VOID, intrinsic);
+    }
```
Comment (tannergooding) [inline src/coreclr/jit/hwintrinsicarm64.cpp:324]: Nit: Add the following to catch things if we have more special cased scalar intrinsics.
```suggestion
assert((category != HW_Category_Scalar) && HWIntrinsicInfo::isScalarIsa(HWIntrinsicInfo::lookupIsa(intrinsic)));
```

**Review by tannergooding — APPROVED** [review]

File: src/coreclr/jit/hwintrinsicarm64.cpp (hunk 1 above)
Comment (echesakovMSFT) [reply src/coreclr/jit/hwintrinsicarm64.cpp:324]: Done.

**Pull request merged**